packages/ember-metal/lib/map.js

```diff
@@ -414,64 +380,61 @@ Map.prototype = {
   @param [options]
     @param {*} [options.defaultValue]
 */
-function MapWithDefault(options) {
-  this._super$constructor();
-  this.defaultValue = options.defaultValue;
-}
+class MapWithDefault extends Map {
+  constructor(options) {
+    super();
+    this.defaultValue = options.defaultValue;
+  }
 
-/**
-  @method create
-  @static
-  @param [options]
-    @param {*} [options.defaultValue]
-  @return {Ember.MapWithDefault|Ember.Map} If options are passed, returns
-    `Ember.MapWithDefault` otherwise returns `Ember.Map`
-  @private
-*/
-MapWithDefault.create = function(options) {
-  if (options) {
-    return new MapWithDefault(options);
-  } else {
-    return new Map();
+  /**
+    @method create
+    @static
+    @param [options]
+      @param {*} [options.defaultValue]
+    @return {Ember.MapWithDefault|Ember.Map} If options are passed, returns
+      `Ember.MapWithDefault` otherwise returns `Ember.Map`
+    @private
+  */
+  static create(options) {
+    if (options) {
+      return new MapWithDefault(options);
+    } else {
+      return new Map();
+    }
   }
-};
 
-MapWithDefault.prototype = Object.create(Map.prototype);
-MapWithDefault.prototype.constructor = MapWithDefault;
-MapWithDefault.prototype._super$constructor = Map;
-MapWithDefault.prototype._super$get = Map.prototype.get;
+  /**
+    Retrieve the value associated with a given key.
 
-/**
-  Retrieve the value associated with a given key.
+    @method get
+    @param {*} key
+    @return {*} the value associated with the key, or the default value
+    @private
+  */
+  get(key) {
+    let hasValue = this.has(key);
 
-  @method get
-  @param {*} key
-  @return {*} the value associated with the key, or the default value
-  @private
-*/
-MapWithDefault.prototype.get = function(key) {
-  let hasValue = this.has(key);
-
-  if (hasValue) {
-    return this._super$get(key);
-  } else {
-    let defaultValue = this.defaultValue(key);
-    this.set(key, defaultValue);
-    return defaultValue;
+    if (hasValue) {
+      return super.get(key);
+    } else {
+      let defaultValue = this.defaultValue(key);
+      this.set(key, defaultValue);
+      return defaultValue;
+    }
   }
-};
 
-/**
-  @method copy
-  @return {Ember.MapWithDefault}
-  @private
-*/
-MapWithDefault.prototype.copy = function() {
-  let Constructor = this.constructor;
-  return copyMap(this, new Constructor({
-    defaultValue: this.defaultValue
-  }));
-};
+  /**
+    @method copy
+    @return {Ember.MapWithDefault}
+    @private
+  */
+  copy() {
+    return copyMap(this, new MapWithDefault({
```

Comment:
Same as above, we need the `this.constructor` stuff

Reply:
thanks good catch 👍 will fix it